### PR TITLE
Integrate Moloch chain events.

### LINKED ISSUE #94: Integrate ChainEvents for Moloch

### PR DESCRIPTION
**Pull request by jnaviask**
## Description
Fixes #94, #243. This PR adds Moloch chain events along with all refactoring necessary to make that happen. Spec and implementation checklist: https://www.notion.so/hicommonwealth/Moloch-Event-Notifications-c5f866195f12481684dcd5672e20c23a

List of changes in PR:
* Implement ChainEvents for Moloch in shared/events and ChainEntity controllers for Moloch events on client, with unit and integration tests. (TODO: increase test coverage?)
* Generalize all server ChainEvent handlers to be chain agnostic! And continue refactoring ChainEvent code in general to be as chain-agnostic as possible.
* Major refactor of Ethereum controllers (accounts/account split into separate files), but still needs more cleanup.
* Turn the small token approval modal into a more serious modal for transferring and approving ERC20s.
* Generalize some aspects of client-side ChainEntity handling by augmenting the controller.
* Completely remove ChainObjects/thegraph dependency, including server routes!

## Motivation and Context
It will permit our users to view and interact with proposals on DAOs.

## How has this been tested?
- Several unit tests for Moloch events.
- Integration test for Moloch events.
- Significant QA against local chain and mainnet.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] yes, but they are not tested because they were deleted

### CODE REVIEW

File: client/scripts/views/modals/token_management_modal.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
+import 'modals/token_management_modal.scss';
+
+import Web3 from 'web3';
+import { default as m } from 'mithril';
```
Comment (drewstone) [inline client/scripts/views/modals/token_management_modal.ts:4]: don't need defaults

File: client/scripts/views/modals/token_management_modal.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import { default as $ } from 'jquery';
```
Comment (drewstone) [inline client/scripts/views/modals/token_management_modal.ts:5]: same here

File: client/scripts/views/modals/token_management_modal.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  toggleValue: ToggleAction;
+  error?: string;
+  recipient?: string;
+  recipientHoldings?: string;
+}
+
+const TokenManagementModal: m.Component<IAttrs, IState> = {
+  oninit: (vnode) => {
+    vnode.state.toggleValue = 'approve';
+    vnode.attrs.account.tokenBalance(vnode.attrs.tokenAddress).then((v) => {
```
Comment (drewstone) [inline client/scripts/views/modals/token_management_modal.ts:34]: we could combine these into `Promise.all` and use one redraw?

File: client/scripts/views/modals/token_management_modal.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    } else {
+      throw new Error(`unknown toggle value: ${vnode.state.toggleValue}`);
+    }
+    return m('.TokenManagementModal', [
+      m('.header', vnode.state.toggleValue === 'approve' ? 'Approve' : 'Transfer'),
+      m('.compact-modal-body', [
+        m('.token-data-label', [ `ERC20 contract address: ${vnode.attrs.tokenAddress}` ]),
+        m('.token-data-label', [ `Your ERC20 holdings: ${vnode.state.tokensAvailable || '--'}` ]),
+        m(RadioSelectorFormField, {
+          callback: async (value: ToggleAction) => {
```
Comment (drewstone) [inline client/scripts/views/modals/token_management_modal.ts:116]: unnecessary async

File: client/scripts/controllers/chain/ethereum/accounts.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+  public init(ChainInfo: EthereumChain): Promise<void> {
+    this._Chain = ChainInfo;
+    return new Promise((resolve, reject) => {
+      // TODO: verify this boilerplate code also works with Ethereum chain API
+      // UPDATE: it did not, so it was delete.
+      // TODO: implement
+
+      // only import bip39 on init to avoid leaking dependencies
+      import('bip39').then(() => resolve());
```
[the remark this answers is not in the document]
Comment (jnaviask) [reply client/scripts/controllers/chain/ethereum/accounts.ts:113]: Yeah -- I didn't know what it did (whether it had some global "on import" effect), so I left it.

**Pull request merged**